src/Microsoft.HttpRepl.Tests/Commands/MockHelpers.cs

```diff
@@ -0,0 +1,39 @@
+using System;
+using Microsoft.Repl;
+using Microsoft.Repl.Commanding;
+using Microsoft.Repl.ConsoleHandling;
+using Microsoft.Repl.Input;
+using Microsoft.Repl.Suggestions;
+using Moq;
+
+namespace Microsoft.HttpRepl.Tests.Commands
+{
+    internal class MockHelpers
+    {
+        internal static IShellState GetMockedShellState(Action<string> writeLineCallback = null, Action<string> errorMessageCallback = null)
```

Comment:
This seems pretty close to already providing a fake implementation of a MockShellState.  You could always have the WriteLineCallback and ErrorMessageCallback exposed as properties, so that tests that use them don't have to set them up with the closure, etc...  just utilize them when needed.

```
var testState = new MockTestState();
...
Assert.Equal(expectedResponseContent, testState.ResponseContent[expectedResponseLines - 2]);
```

Reply:
This is done now.